gormtestsupport/benchmark/db_bench_suite.go

```diff
@@ -0,0 +1,73 @@
+package benchmark
+
+import (
+	"os"
+
+	config "github.com/almighty/almighty-core/configuration"
+	"github.com/almighty/almighty-core/log"
+	"github.com/almighty/almighty-core/migration"
+	"github.com/almighty/almighty-core/models"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/almighty/almighty-core/test"
+	"github.com/almighty/almighty-core/workitem"
+	"github.com/jinzhu/gorm"
+	_ "github.com/lib/pq" // need to import postgres driver
+
+	"golang.org/x/net/context"
+)
+
+var _ test.SetupAllSuite = &DBBenchSuite{}
+var _ test.TearDownAllSuite = &DBBenchSuite{}
+
+// NewDBBenchSuite instanciate a new DBBenchSuite
+func NewDBBenchSuite(configFilePath string) DBBenchSuite {
+	return DBBenchSuite{configFile: configFilePath}
+}
+
+// DBBenchSuite is a base for tests using a gorm db
+type DBBenchSuite struct {
+	test.Suite
+	configFile    string
+	Configuration *config.ConfigurationData
+	DB            *gorm.DB
+}
+
+// SetupSuite implements suite.SetupAllSuite
+func (s *DBBenchSuite) SetupSuite() {
+	resource.Require(s.B(), resource.Database)
+	configuration, err := config.NewConfigurationData(s.configFile)
+	if err != nil {
+		log.Panic(nil, map[string]interface{}{
+			"err": err,
+		}, "failed to setup the configuration")
+	}
+	s.Configuration = configuration
+	if _, c := os.LookupEnv(resource.Database); c != false {
+		s.DB, err = gorm.Open("postgres", s.Configuration.GetPostgresConfigString())
+		if err != nil {
+			log.Panic(nil, map[string]interface{}{
+				"err":             err,
+				"postgres_config": configuration.GetPostgresConfigString(),
+			}, "failed to connect to the database")
+		}
+	}
+}
+
+// PopulateDBBenchSuite populates the DB with common values
+func (s *DBBenchSuite) PopulateDBBenchSuite(ctx context.Context) {
```

Comment:
I'd say, embed the DBTestSuite struct in your DBBenchSuite struct for now and don't duplicate the code. Then you have a separation of types but you can still re-use the functions.